### DISCUSSION TITLE
"Naked" tabularray

### DISCUSSION THREAD

**Discussion by Rmano** [lvjr/tabularray#535] [Q&A]
I'd like to create tabularray that are to be considered part of the text --- not floating (ok, that's easy), and not interfering with the counting of the tables that are floating in the document. Although I managed to remove the caption, the counter is still ticking... see:

```latex
\documentclass{article}
\usepackage[T1]{fontenc}
\usepackage{tabularray}
\NewTblrTheme{mystyle}{
    \SetTblrStyle{head}{\bfseries}%{font=\bfseries} does not work
    \DefTblrTemplate{caption}{default}{}
}
\begin{document}
\begin{longtblr}[
    theme=mystyle,
    ]{%
        colspec={X[4,r]X[1,c]X[1,l]X[3,l]X[1.5,r]},
        rowhead=1, rowfoot=0, hlines, vlines,
    }
    1 & 2 & 3 & 4 & 5 \\
    1 & 2 & 3 & 4 & 5 \\
    1 & 2 & 3 & 4 & 5 \\
\end{longtblr}

\begin{table}[htb]
    \caption{First floating table}
    \begin{center}
        Content here
    \end{center}
\end{table}
\end{document}
```

![image](https://github.com/user-attachments/assets/be465751-8ab2-4fdd-82ad-cb4079dd349d)

The table is called "Table 2" in the document.

Is this possible? Am I overlooking something in the manual?

**Comment by muzimuzhi** [comment]
`\begin{longtblr}[entry=none,label=none]`, see package doc (v2024A, 2024-02-16), p. 39 (end of sec. 4.1).

Similar: https://github.com/lvjr/tabularray/discussions/285#discussioncomment-3381881 and https://github.com/lvjr/tabularray/discussions/334#discussioncomment-4384372

**Comment by Rmano** [comment]
Thanks! Sorry, I missed them. Is it possible to add them in a theme definition?

**Reply by muzimuzhi** [reply]
Interesting question. Checking how `\l__tblr_table_no_label_bool` is set and used, it seems the answer is no.

**Reply by lvjr** [reply]
To avoid typing them every time, you can write `\SetTblrOuter[longtblr]{entry=none,label=none}`.

**Reply by Rmano** [reply]
Thanks for the hint!

Maybe this could be a "feature request", but... wouldn't be nice if one could do:

```latex
\NewTblrTheme{mystyle}{
    \SetTblrStyle{head}{\bfseries}
    \SetTblrInner{rowhead=1, rowfoot=0, hlines, vlines,
    row{1}={font=\bfseries},}
    \SetTblrOuter{ label=none, entry=none,}
}
\begin{document}
\begin{longtblr}[theme=mystyle] {X[4,r]X[1,c]X[1,l]X[3,l]X[1.5,r]}
    1 & 2 & 3 & 4 & 5 \\
    1 & 2 & 3 & 4 & 5 \\
[...]
```

...and have a self-contained theme for everything? Similar to what you can do with, say, `\pgfplotsset{aaa/.style={...}}`? 

Thanks anyway for the nice package!

**Reply by lvjr** [reply]
The following example shows the recommented way in `tabularray` to achieve this:
```latex
\documentclass{article}
\usepackage{tabularray}
\NewTblrTheme{mystyle}{
  \SetTblrStyle{head}{\bfseries}
}
\NewTblrEnviron{mylongtblr}
\SetTblrInner[mylongtblr]{rowhead=1, rowfoot=0, hlines, vlines, row{1}={font=\bfseries}}
\SetTblrOuter[mylongtblr]{long, label=none, entry=none, theme=mystyle}
\begin{document}
\begin{mylongtblr}{
  colspec={X[4,r]X[1,c]X[1,l]X[3,l]X[1.5,r]}
}
  1 & 2 & 3 & 4 & 5 \\
  1 & 2 & 3 & 4 & 5 \\
  1 & 2 & 3 & 4 & 5 \\
\end{mylongtblr}
\end{document}
```

**Reply by Rmano** [reply]
Yes, I tried it, but... it seems that  `mylongtblr` is not a breakable table...  (try to repeat the row 40-50 times).

Is there something like `\NewLongTblrEnviron` or similar?

**Reply by lvjr** [reply]
Oh, I forgot to add `long` option. The example has been updated.

**Reply by Rmano** [reply]
Wow, thanks! Now it's perfect.
